oss-internship-2020/pffft/README.md

```diff
@@ -0,0 +1,73 @@
+# Sandboxing PFFFT library
+
+Builder: CMake  
+OS: Linux
+
+### For testing: 
+`cd build`, then `./pffft_sandboxed`
+
+### For debug:
+`SAPI_VLOG_LEVEL=1 ./pffft_sandboxed --v=100
+--sandbox2_danger_danger_permit_all_and_log <auxiliar file>`
+
+## ***About the project*** 
+*PFFFT library is concerned with 1D Fast-Fourier Transformations finding a
+compromise between accuracy and speed. It deals with real and complex
+vectors, both cases being illustrated in the testing part (`main_pffft.c` 
+for initially and original version, `main_pffft_sandboxed.cc` for our 
+currently implemented sandboxed version).
+The original files can be found at: https://bitbucket.org/jpommier/pffft/src.*
+
+*The purpose of sandboxing is to limit the permissions and capabilities of 
+library’s methods, in order to secure the usage of them. 
+After obtaining the sandbox, the functions will be called through an 
+Sandbox API (being called `api` in the current test) and so, the 
+operations, system calls or namspaces access may be controlled. 
+From both `pffft.h` and `fftpack.h` headers, useful methods are added to 
+sapi library builded with CMake. There is also a need to link math library 
+as the transformations made require mathematical operators. 
+Regarding the testing of the methods, one main is doing this job by 
+iterating through a set of values, that represents the accuracy of 
+transformations and print the speed for each value and type of 
+transformation. More specifically, the input length is the target for 
+accuracy (named as `N`) and it stands for the number of data points from 
+the series that calculate the result of transformation. It is also 
+important to mention that the `cplx` variable stands for a boolean value 
+that tells the type of transformation (0 for REAL and 1 for COMPLEX) and 
+it is taken into account while testing.
+In the end, the performance of PFFFT library it is outlined by the output.*
+
```

Comment:
Add a section that explains how to check out the code for PFFFT and how to invoke CMake with your integration.

For example, the commands to check out and build could look something like this:
```
mkdir -p build && cd build
git clone https://bitbucket.org/jpommier/pffft.git
cmake .. -G Ninja -DPFFFT_ROOT_DIR=$PWD
ninja
```